ci/images/ci-runner/hack/sidecar/bin/plnsvc_setup.sh

```diff
@@ -11,6 +11,10 @@ REPO_REVISION=$2
 # to avoid the issue of "fatal: detected dubious ownership in repository at '/workspace/source'"
 git config --global --add safe.directory "$PWD"
 
+# Checkout the branch we want to setup
+git fetch origin "$REPO_REVISION"
+git checkout "$REPO_REVISION"
```

Comment:
Honestly, I prefer using git-clone task as a common task to handle it

Reply:
My understanding is that using git-clone is inefficient.
* We'll need to clear the destination then copy the whole content over.
* I believe the directory might be deleted before the clone, so we're going to effectively clone the repo 3 times instead of one.